lib/matplotlib/path.py

```diff
@@ -136,21 +136,26 @@ def __init__(self, vertices, codes=None, _interpolation_steps=1,
         else:
             vertices = np.asarray(vertices, np.float_)
 
+        if (vertices.ndim != 2) or (vertices.shape[1] != 2):
+            msg = "'vertices' must be a 2D list or array with shape Nx2"
+            raise ValueError(msg)
+
         if codes is not None:
             codes = np.asarray(codes, self.code_type)
-            assert codes.ndim == 1
-            assert len(codes) == len(vertices)
-            if len(codes):
-                assert codes[0] == self.MOVETO
+            if (codes.ndim != 1) or len(codes) != len(vertices):
```

Comment:
nope: `assert a == 1; assert b == 2` is equivalent to `assert a == 1 and b ==2`. In the `if` the condition is negated, so it becomes `a != 1 or b != 2`